### ISSUE TITLE
Site Fields which are Lookup Fields are created before the List which is to be used in the lookup have not been created yet. 

### ISSUE THREAD

**Issue by NPrice99** [OfficeDev/PnP#906]
When a site field Is created and the list that is to be used for the lookup has not yet been created, results in an invalid field where the ListName is blank.(see attached file)
eg When I create a lookup field using the provisioning engine, this is the XML which is generated :- <Field Type="Lookup" DisplayName="Directorate" Required="FALSE" EnforceUniqueValues="FALSE" List="Lists/Directorate" WebId="59eea740-3e15-49b8-942e-78f890024f6f" ShowField="Title" UnlimitedLengthInDocumentLibrary="FALSE" Group="My Columns" ID="{3e39d7cd-9d83-42de-b15c-a1f54610e91a}" SourceID="{59eea740-3e15-49b8-942e-78f890024f6f}" StaticName="Directorate" Name="Directorate" />  NB the List is Lists/Directorate.  The attached file shows the resultant field.  Should there not be an error message generated when this happens ?

![lookupfieldsevidence](https://cloud.githubusercontent.com/assets/8484048/8057890/2f3dde6c-0ead-11e5-8aa1-1729a24cfe39.png)

**Comment by braegelno5** [comment]
at the moment all lists are saved to the template not in order on dependencies to each other
we should implement this
we should check this in ProvisioningTemplate and also in ProvisionObjects

**Comment by PaoloPia** [comment]
We do support lookup fields and we provision lookup fields after lists, so that we can find the source list. Are you still experiencing issues with this? Have you seen this sample template?

https://github.com/OfficeDev/PnP-Provisioning-Schema/blob/master/Samples/LookupField.xml

**Comment by NPrice99** [comment]
Hi Paolo The issue is you need to define fields before you can define Lists, and this is the order the provisioning engine does things in.  However, there are lookup fields which need lists to be available before the field can be created and this is the problem.  We almost need the provisioning engine to do two passes at the template.  One to create all of the fields which are not reliant on a list being there and a second pass when all of the lists are there, to create all of the lookup fields which are dependant on there being a list being there so that list can be used to look things up in.

**Comment by braegelno5** [comment]
I have run an debug test against this problem, nprice99 makes the point clear, that's why I wrote we must implement this, it's not yet implemented but in my head..

**Comment by braegelno5** [comment]
So the update for loockup field was implemented as an object event, but it is not working yet, maybe there we need also an update for the web I'd.. Digging

**Comment by erwinvanhunen** [comment]
What we do with lookup fields is as follows: 

1. We create the field, however, we do _not_ connect it yet to the list it looks up data from
2. We create the lists.
3. We work our way again through the list of the fields in the template, if a lookup field is present, we then find the field that was created in step 1 and we update the entry to connect to the list created in step 2

If there are other cases where a field depends on a list to be present before the field is created, I would love to see examples of that.

**Comment by NPrice99** [comment]
Hi Erwin

I have seen a number of instances where the field has been created but the list has not been connected to the field even though the List itself has been created.

I cannot think of any other instance where a field depends upon a list being present.

Regards

Nigel

**Comment by erwinvanhunen** [comment]
If such an issue occurs, the thing that could help us a lot is get detailed information about the template being used (e.g. the XML).

To @braegelno5 : what kind of dependencies do you define from one list to the other?

**Comment by NPrice99** [comment]
Hi Erwin an example of a lookup field without a list to lookup, is shown in the original post on this issue.

**Comment by NPrice99** [comment]
![branchevidence](https://cloud.githubusercontent.com/assets/8484048/8185501/bcdbf49a-143c-11e5-88e8-51ea3be21f24.jpg)

        <Field Type="Lookup" DisplayName="Branch" Required="FALSE" EnforceUniqueValues="FALSE" List="Lists/Branch" WebId="59eea740-3e15-49b8-942e-78f890024f6f" ShowField="Title" UnlimitedLengthInDocumentLibrary="FALSE" Group="DH Columns" ID="{904ffc10-12cb-4b57-8311-0ed34e8cc447}" SourceID="{59eea740-3e15-49b8-942e-78f890024f6f}" StaticName="Branch" Name="Branch" />
 
       <pnp:ContentType ID="0x0100A9AB2FB55EA60448AC35A507D73B687B" Name="DH_Branch_Item" Description="" Group="DH Content Types">
          <pnp:FieldRefs>
            <pnp:FieldRef ID="c042a256-787d-4a6f-8a8a-cf6ab767f12d" Name="ContentType" />
            <pnp:FieldRef ID="fa564e0f-0c70-4ab9-b863-0177e6ddd247" Name="Title" Required="true" />
            <pnp:FieldRef ID="f3b0adf9-c1a2-4b02-920d-943fba4b3611" Name="TaxCatchAll" Hidden="true" />
            <pnp:FieldRef ID="8f6b6dd8-9357-4019-8172-966fcd502ed2" Name="TaxCatchAllLabel" Hidden="true" />
            <pnp:FieldRef ID="3e39d7cd-9d83-42de-b15c-a1f54610e91a" Name="Directorate" />
            <pnp:FieldRef ID="d47b2e15-86da-49d4-ac45-acb64028bad7" Name="Division" />
            <pnp:FieldRef ID="1ee6afac-97da-423a-9ca4-1669ddecdc28" Name="Branch_x0020_Code" />
          </pnp:FieldRefs>
        </pnp:ContentType>
 
 
<pnp:ListInstance Title="Branch" Description="" DocumentTemplate="" TemplateType="100" Url="Lists/Branch" MinorVersionLimit="0" MaxVersionLimit="0" TemplateFeatureID="00bfea71-de22-43b2-a848-c05709900100" ContentTypesEnabled="true" EnableAttachments="false" EnableFolderCreation="false">
          <pnp:ContentTypeBindings>
            <pnp:ContentTypeBinding ContentTypeID="0x0100A9AB2FB55EA60448AC35A507D73B687B" Default="true" />
          </pnp:ContentTypeBindings>
          <pnp:Views>
            <View Name="{6360AFF8-750D-4B16-A2DB-CF88EEE7C1ED}" DefaultView="TRUE" MobileView="TRUE" MobileDefaultView="TRUE" Type="HTML" DisplayName="All Items" Url="/sites/lookup/Lists/Branch/AllItems.aspx" Level="1" BaseViewID="1" ContentTypeID="0x" ImageUrl="/_layouts/15/images/generic.png?rev=23">
              <Query>
                <OrderBy>
                  <FieldRef Name="ID" />
                </OrderBy>
              </Query>
              <ViewFields>
                <FieldRef Name="LinkTitle" />
                <FieldRef Name="Branch_x0020_Code0" />
                <FieldRef Name="Directorate0" />
              </ViewFields>
              <RowLimit Paged="TRUE">150</RowLimit>
              <Aggregations Value="Off" />
              <JSLink>clienttemplates.js</JSLink>
              <XslLink Default="TRUE">main.xsl</XslLink>
              <Toolbar Type="Standard" />
            </View>
          </pnp:Views>
          <pnp:FieldRefs>
            <pnp:FieldRef ID="3e39d7cd-9d83-42de-b15c-a1f54610e91a" Name="Directorate0" DisplayName="Directorate" />
            <pnp:FieldRef ID="d47b2e15-86da-49d4-ac45-acb64028bad7" Name="Division0" DisplayName="Division" />
            <pnp:FieldRef ID="1ee6afac-97da-423a-9ca4-1669ddecdc28" Name="Branch_x0020_Code0" DisplayName="Branch Code" />
          </pnp:FieldRefs>
        </pnp:ListInstance>

**Comment by braegelno5** [comment]
I have now a new PR to fix that. The WebId was not updated in the actual code, so if we run the template on another sitecollection, it was not enough to update the List with guid, we need also the guid for parent web from list to the new one.

**Comment by NPrice99** [comment]
Sorry I forgot to mention the source and destination of the above example are in different site collections.  Is the fix in the dev branch now ?

**Comment by braegelno5** [comment]
The PR #944 is at this moment open

**Comment by NPrice99** [comment]
braegeIno5 - I tried this and I still do not get the ListName in the lookup field.

**Comment by braegelno5** [comment]
You have used the not merged file into dev the actual dev branch from #944 on your own clone? I'm thinking on rewriting the fix, so we will pull the web Id from existing list parent web..

**Comment by braegelno5** [comment]
On my test tenant it is working on creating the List with sc lookup colum.
But the list handler on a second run fails now with  field Id error..

**Comment by NPrice99** [comment]
yes I used the file which has not been merged yet into my own clone.  My issue has always been that the list to be used as a lookup is always blank in the lookup field.

**Comment by braegelno5** [comment]
fix #944 depend on #929, i should create a patch for both?
the update field in object list instance will fail without #929 in an second template update run, because teh in the patch removed web.Fields.GetById(fieldId) throws now an error for the field 'Invalid field name', maybe because the field was properly working.. update with List attribute removing bevore

**Comment by braegelno5** [comment]
have done new summary pull request #953

**Comment by NPrice99** [comment]
Hi Erwin and Torsen I am afraid it failed on my on-premises farm :- 
![latest-error](https://cloud.githubusercontent.com/assets/8484048/8219025/000e2a0a-153f-11e5-81d3-edd95a77424d.jpg)

This looks similar to the one I fixed on my copy of the dev branch last night. Issue #950.

I added AsEnumerable(). so that the line of code looks like this.
So List<Guid> targetIds = existingContentType.FieldLinks.AsEnumerable().Select(c1 => c1.Id).ToList();

Regards

Nigel

**Comment by erwinvanhunen** [comment]
Nigel,

The provisioning engine is mainly tested against O365 and that's where most of the focus lies right now. While we do want to support on-premises as soon as possible, it still requires way more testing from our side too. 

On-prem client SDK lacks certain implementations when compared to the O365 version of the SDK. A solution is indeed to use AsEnumerable(). 

I'll fix this one, but you're more than welcome to fix other issues too and submit them as pull requests.

**Comment by NPrice99** [comment]
Hi Erwin

Yes, I know that the focus is on O365.  That’s why Vesa asked me to join the party.  I am here to test against on-premises.

I would not trust my programming as  far as pull requests, but If I find a problem and a solution, I will let you know and you can check my programming before a pull request is created.

Thanks for all of your help again.

Regards

Nigel

**Comment by NPrice99** [comment]
After fixing above error - it went on a long way until it got to creating list instances and then crashed again :- 
![latest-error](https://cloud.githubusercontent.com/assets/8484048/8219839/c4804c06-1544-11e5-8826-5ef41f6b195a.jpg)

NB I have recompiled OfficeDevPnP.core and I have recompiled and rerun my C# program which calls web.ApplySPOProvisioningTemplate.  I have not rerun my C# program to call Get-SPOProvisioningTemplate.  That I will do in the morning.

**Comment by erwinvanhunen** [comment]
I fixed the first issue you mentioned in the dev branch now, The next crash you encounter, is that one the same (fixed?) line?

**Comment by erwinvanhunen** [comment]
I'm closing this issue now as the current issues you encounter do not seem to be related to the initial issue. Please correct me if I'm wrong and I'll reopen the issue.

**Comment by NPrice99** [comment]
No - its a lot further on when provisioning the List Instances and checking to see if we need to add a content type.

**Comment by NPrice99** [comment]
Yes they are still related to the initial issue.  I am still getting lookup fields without a list to lookup as per the first posting on this issue and further down where I show the lookup field without the list to lookup and the XML which was used to generate it.

**Comment by erwinvanhunen** [comment]
Can you mail me your full xml template?

**Comment by NPrice99** [comment]
On its way

**Comment by NPrice99** [comment]
Just tested latest version in dev and it failed on line 602 of ObjectListInstance.cs (ie It got further than last time) "Invalid Template URL".web.Context.Load(createdList.ContentTypes); web.Context.ExecuteQueryRetry();

**Comment by braegelno5** [comment]
on line 602 (https://github.com/OfficeDev/PnP/blob/dev/OfficeDevPnP.Core/OfficeDevPnP.Core/Framework/Provisioning/ObjectHandlers/ObjectListInstance.cs#L602) ?

**Comment by NPrice99** [comment]
Hi Torsten whats up? That's where it failed.

**Comment by braegelno5** [comment]
you mean 601 but failed on execution 602; maybe the same like before in update fields (the url was set for lookupfield -- works; and a second run removes the working Url from xml schema..) ..digging how to solve here

**Comment by braegelno5** [comment]
my first suggestion for this: we should rewrite all the List url  removing, so we also remove the Webid and SourceId, or we must check if the list on url exists in web with id..

**Comment by braegelno5** [comment]
In my test settings this error not occurs, what must i set in ct so I'm able to check my changes?

**Comment by NPrice99** [comment]
I am using on-premises I suspect you and Erwin are using O365

**Comment by braegelno5** [comment]
So my suggestion is: on the beginning we have on prem and o365 with removing the URL - 'only' a not working lookupfield, the WebId and SourceId was set but was not associated to content db. I think the wrong id's will give null for web return so the List will not checked and throws no error. So we should change the code, to set also the id's to Guid.Empty on the first creation and update processing in objectlistinstance.

**Comment by braegelno5** [comment]
code to check against on prem:

Line 280 et seq.: private static void CreateField(XElement fieldElement, ListInfo listInfo)
        {
            TemporaryRemoveListAsscociation(fieldElement);

            var fieldXml = fieldElement.ToString();
            listInfo.SiteList.Fields.AddFieldAsXml(fieldXml, false, AddFieldOptions.DefaultValue);
        }

        private static void TemporaryRemoveListAsscociation(XElement fieldElement)
        {
            var attributeNameArray = new[] { "List", "WebId", "SourceID" };

            foreach (var name in from name in attributeNameArray let identifier = fieldElement.Attribute(name) != null ? fieldElement.Attribute(name).Value : null where identifier != null select name)
            {
                // Temporary remove attribute from fieldElement
                fieldElement.Attribute(name).Remove();
            }
        }

Line 314 TemporaryRemoveListAsscociation(fieldElement);

**Comment by NPrice99** [comment]
Just tested latest version of GetProvisioningTemplate and get error in Line 686 of ObjectListInstance.cs.  The foreach (var item in lists.Where(l => l.hidden == false) needs a AsEnumerable() between lists and Where.  Also line 754 of ObjectListInstance.cs foreach( (var view in item.views.Where(view => !view.Hidden)) needs a AsEnumerable() between View sand Where.  Also line 763 between Fields and Where. These changes seem to have worked.
Hi Erwin - if you let know how to set up pull requests, I can let you have my version of Ob jectListInstance.cs

**Comment by NPrice99** [comment]
Still no sign of a list in Lookup Fields yet.  Latest version crashes when adding a Content Type to a List.  (FieldandContentTypeExtensions.cs line 667) I will try Torsten's changes in the morning

**Comment by braegelno5** [comment]
If this List.Hidden in where clause of Linq to object not work on sp15 I suggest we should rewrite them with compiler switch #if SP15Target.. or we remove them. It was added to limit the network flow for all lists from server to client

**Comment by braegelno5** [comment]
But again a general  question: as the issue was opened in the beginning.. there was in the end an ct with no list associated.
Was the PS or whatever you are running on sp15 running twice over the same target site manually and all was fine expect the lookup ct with no list assoc?

**Comment by NPrice99** [comment]
I was running a C# program - No there was a crash 80070057 in FieldandContentypeExtensions.cs line 667

**Comment by braegelno5** [comment]
What i mean, in the beginning 13 Days before, here on top of this discussion, there you was able to run to the end without this error, only the ct was not properly set, so a list using this ct was crashing in the ui? And second i you was running the C# programm a second time you finished also without an error?

**Comment by NPrice99** [comment]
In the beginning, 13 days ago, I could run the Apply Templates all of the way through, using a C# programme.  Since then, on the DEV branch, I have not been able to run all of the way through as I keep hitting other errors.  The original fault was that on a lookup field, the list to be looked up was not being populated.

**Comment by braegelno5** [comment]
ok and than my next question for beginning: the generated template had also a WebId for this LookupField from the source.  And next this id was not in the contentdb where the target sitecollection lives in?

**Comment by NPrice99** [comment]
![80057007](https://cloud.githubusercontent.com/assets/8484048/8279063/d41e2cd2-18c8-11e5-8b82-8e2d6493595b.png)
The 80057007 error also occurs when using the GUI.  screenshot enclosed, so it looks like its an issue with the Content Type.

**Comment by braegelno5** [comment]
with begin i meant before 13 days on start this threat. ..because there was only the list attribute not set.. web and source id was set to the source web id by default. that's i query for the contentdb <-> sc

**Comment by NPrice99** [comment]
The content db is deleted and re-created each time I run the program

**Comment by braegelno5** [comment]
the only what i can do at the moment  is: i make a new PR against the actual dev branch, you pick the three files up from the ui view and test it on your on prem sp.
But before i will do this, i will completely understood your error (without debugging the on prem sp code byself)..
at the beginning we have a engine where all lookup fields have a kosher List attribute. The Web and Source id attribute was not reset to an id from parent web of this list. On your system and also on o365 this works (fine). In the end we have a not kosher lookup field - we are able to open the field in the ui but there is no list shown. If this field is associated to a list we can't open the default view for this list without an exception.
That's correct?
The field was not kosher set, because the WebId and SourceId was not set to the parent web from the List assigned to a web relative url.
After this was detected as the daemon in this process we added some lines of code in the end where not only the List is set to the url, but also the WebId and SourceId to the parent web id of the list.
And now it works on o365 in the right manner, in the end we have a list where the lf in default view throws no exception, all was fine.
But on OnPrem you get errors. I suggest the AsEnumerable() you have added to your code so the project compiles and then it crashes if the new ct (with at least one at this moment not kosher lookup field) is added to the list.
But why, is there a change in the flow.. digging..

**Comment by NPrice99** [comment]
Hi Torsten

I am also digging. It looks as if there may be some wrongly generated XML in a Content Type  in the template.  This may explain the 80070057 error.

I will keep you posted.

Regards

Nigel

**Comment by NPrice99** [comment]
Hi - I have made some progress. There was one content type which was generating extra fields in the Template, compared with the Content Type definition in the GUI.  However, if I ran SPManager, I can see two extra fields which I had deleted some time ago.  I could not delete them using SPManager, So I got out some PowerShell which did delete them.  I ran Get Template, and the extra fields were not there.  I ran Apply Template and it got further (no 80057007) but failed when processing "Site Assets" - Invalid Template URL In ObjectListInstance.cs line 602.  I think it is time to put Torsten's changes in.  Lookup fields still have the List entry blank.

**Comment by NPrice99** [comment]
Hi Torsten  I am still crashing in ObjectListance now Line 610 because of the recent additions with "Invalid Template Url" Here ->  web.Context.Load(createdList.Views);
            web.Context.Load(createdList, l => l.Id);
            web.Context.Load(createdList, l => l.RootFolder.ServerRelativeUrl);
            web.Context.Load(createdList.ContentTypes);

The Lookup Field's List is still blank (Probably because we have not created the list yet ) and we proabably not gone around a second time yet to fix the List up.
            web.Context.ExecuteQueryRetry(); <-------------------------------------------------------

Whatever, Invalid Template url means.

**Comment by NPrice99** [comment]
Hi - I have found the cause of the Invalid Template Url - in the Provisioning template, the list on which it failed on was the only one which had DocumentTemplate="{site}/SiteAssets/Forms/template.doc".  I set this to DocumentTemplate="" and the problem went away.  There seems to be lots of issues around Community lists.  80070057 - I just deleted them from the Provisioning template as I did not need them.  I have now hit a new problem "Cannot change Hidden attribute for this field".  This has stumped me because it is "CheckOutUser" which is a ReadOnly, Hidden field which is an OOTB standard field. 

 I cannot see how Lists within Lookup Fields can work :-: See below - the Listname on the lookup field is a GUID which is not relevant to the new site.  The List Instance itself does not have the GUID only a name.
<Field Type="Lookup" DisplayName="Directorate" Required="FALSE" EnforceUniqueValues="FALSE" List="78f760d7-0fbc-491d-a7fe-b5beffa56f09" ShowField="Title" UnlimitedLengthInDocumentLibrary="FALSE" RelationshipDeleteBehavior="None" ID="{fc2420f6-a399-405d-9dba-d0b3c6b9eee3}" SourceID="{f3f1caa8-f0d9-46bf-a290-eab6e5c3a281}" StaticName="Directorate" Name="Directorate" ColName="int1" RowOrdinal="0" Version="6" />
            <Field Type="Lookup" DisplayName="Division" Required="FALSE" EnforceUniqueValues="FALSE" List="e391a904-bb54-4266-b435-f5127d37e992" ShowField="Title" UnlimitedLengthInDocumentLibrary="FALSE" RelationshipDeleteBehavior="None" ID="{35545ed6-269e-4878-8799-19dac9275504}" SourceID="{f3f1caa8-f0d9-46bf-a290-eab6e5c3a281}" StaticName="Division" Name="Division" ColName="int2" RowOrdinal="0" Version="6" />
            <Field Type="Lookup" DisplayName="Branch" Required="FALSE" EnforceUniqueValues="FALSE" List="802081f1-5a67-4414-a7c4-cafbe9ded3d7" ShowField="Title" UnlimitedLengthInDocumentLibrary="FALSE" RelationshipDeleteBehavior="None" ID="{428c746d-2e41-47ce-b014-85c5406b73da}" SourceID="{f3f1caa8-f0d9-46bf-a290-eab6e5c3a281}" StaticName="Branch" Name="Branch" ColName="int3" RowOrdinal="0" Version="6" />

     <pnp:ListInstance Title="Directorate" Description="" DocumentTemplate="" TemplateType="100" Url="Lists/Directorate" MinorVersionLimit="0" MaxVersionLimit="0" TemplateFeatureID="00bfea71-de22-43b2-a848-c05709900100" ContentTypesEnabled="true" EnableAttachments="false" EnableFolderCreation="false">
         <pnp:ListInstance Title="Division" Description="" DocumentTemplate="" TemplateType="100" Url="Lists/Division" MinorVersionLimit="0" MaxVersionLimit="0" TemplateFeatureID="00bfea71-de22-43b2-a848-c05709900100" ContentTypesEnabled="true" EnableAttachments="false" EnableFolderCreation="false">
        <pnp:ListInstance Title="Branch" Description="" DocumentTemplate="" TemplateType="100" Url="Lists/Branch" MinorVersionLimit="0" MaxVersionLimit="0" TemplateFeatureID="00bfea71-de22-43b2-a848-c05709900100" ContentTypesEnabled="true" EnableAttachments="false" EnableFolderCreation="false">

**Comment by NPrice99** [comment]
The formatting left out the field markup - 
    <Field Type="Lookup" DisplayName="Directorate" Required="FALSE" EnforceUniqueValues="FALSE" List="78f760d7-0fbc-491d-a7fe-b5beffa56f09" ShowField="Title" UnlimitedLengthInDocumentLibrary="FALSE" RelationshipDeleteBehavior="None" ID="{fc2420f6-a399-405d-9dba-d0b3c6b9eee3}" SourceID="{f3f1caa8-f0d9-46bf-a290-eab6e5c3a281}" StaticName="Directorate" Name="Directorate" ColName="int1" RowOrdinal="0" Version="6" />
            <Field Type="Lookup" DisplayName="Division" Required="FALSE" EnforceUniqueValues="FALSE" List="e391a904-bb54-4266-b435-f5127d37e992" ShowField="Title" UnlimitedLengthInDocumentLibrary="FALSE" RelationshipDeleteBehavior="None" ID="{35545ed6-269e-4878-8799-19dac9275504}" SourceID="{f3f1caa8-f0d9-46bf-a290-eab6e5c3a281}" StaticName="Division" Name="Division" ColName="int2" RowOrdinal="0" Version="6" />
            <Field Type="Lookup" DisplayName="Branch" Required="FALSE" EnforceUniqueValues="FALSE" List="802081f1-5a67-4414-a7c4-cafbe9ded3d7" ShowField="Title" UnlimitedLengthInDocumentLibrary="FALSE" RelationshipDeleteBehavior="None" ID="{428c746d-2e41-47ce-b014-85c5406b73da}" SourceID="{f3f1caa8-f0d9-46bf-a290-eab6e5c3a281}" StaticName="Branch" Name="Branch" ColName="int3" RowOrdinal="0" Version="6" />

**Comment by NPrice99** [comment]
<Field Type="Lookup" DisplayName="Directorate" Required="FALSE" EnforceUniqueValues="FALSE" List="78f760d7-0fbc-491d-a7fe-b5beffa56f09" ShowField="Title" UnlimitedLengthInDocumentLibrary="FALSE" RelationshipDeleteBehavior="None" ID="{fc2420f6-a399-405d-9dba-d0b3c6b9eee3}" SourceID="{f3f1caa8-f0d9-46bf-a290-eab6e5c3a281}" StaticName="Directorate" Name="Directorate" ColName="int1" RowOrdinal="0" Version="6" />
            <Field Type="Lookup" DisplayName="Division" Required="FALSE" EnforceUniqueValues="FALSE" List="e391a904-bb54-4266-b435-f5127d37e992" ShowField="Title" UnlimitedLengthInDocumentLibrary="FALSE" RelationshipDeleteBehavior="None" ID="{35545ed6-269e-4878-8799-19dac9275504}" SourceID="{f3f1caa8-f0d9-46bf-a290-eab6e5c3a281}" StaticName="Division" Name="Division" ColName="int2" RowOrdinal="0" Version="6" />
            <Field Type="Lookup" DisplayName="Branch" Required="FALSE" EnforceUniqueValues="FALSE" List="802081f1-5a67-4414-a7c4-cafbe9ded3d7" ShowField="Title" UnlimitedLengthInDocumentLibrary="FALSE" RelationshipDeleteBehavior="None" ID="{428c746d-2e41-47ce-b014-85c5406b73da}" SourceID="{f3f1caa8-f0d9-46bf-a290-eab6e5c3a281}" StaticName="Branch" Name="Branch" ColName="int3" RowOrdinal="0" Version="6" />

**Comment by braegelno5** [comment]
This Hidden field I'm also twice seen, but i was not able to reproduce this, so you have it also.
We should at first check the attribute with equalnes and update/set only if there is a diff..

**Comment by braegelno5** [comment]
i have no idea, the error will not arise again on light..
in my template are three site fields with ReadOnly="TRUE", one is the additional field for a lookup column in my ct and the others are SharedWithUsers and SharedWithDetails, fields from std sp site #0 but not in our base template, but again and again all went fine on o365
and the question should be what the hell means "Cannot change Hidden attribute for this field", because i think there is no change only an reset from true to true

**Comment by NPrice99** [comment]
With the latest get from the dev branch I can now get to the end of "Apply Template" (All 13 phases of it). However, the lookup fields still don't have a List Name in them.  The lookuplist attribute of the lookup field has a GUID in it which from SharePoint Manager is the GUID of the correct list.  SO I don't understand why the GUI is not showing a list from where the information is supposed to come from.

**Comment by NPrice99** [comment]
This looks just like someone has deleted the field (or a list containing a field) and it has been restored from backup and the Lookup list has become disconnected from the lookup site column.

**Comment by braegelno5** [comment]
The List property in template should be a web relative URL, with this the list will found on target system, if found the URL will changed to the id of the list. But also should the WebId and SourceId set to an I'd. Both id's must be the parent web id where the list is hosted.

**Comment by NPrice99** [comment]
Hi Torsten / Erwin
I can confirm that if I run the following powershell script on Lookup fields created by Apply-SPOProvisioningTemplate they lookup list reappears and can be used.

![fixlookupfields](https://cloud.githubusercontent.com/assets/8484048/8500734/d962811c-2196-11e5-85ab-ab3ca1c43f51.JPG)

**Comment by NPrice99** [comment]
Here is before and afterwards -
![beforeandafterwards](https://cloud.githubusercontent.com/assets/8484048/8500867/15ef30de-2198-11e5-8a90-31cb4ad6209d.jpg)
Look at the listID before no "{" and afterwards "{" could this be our problem ?

**Comment by NPrice99** [comment]
I have just gone into SharePoint Manager and found a lookup field which was not showing the lookup list, add"{" at the front and "}" at the back and hey presto ! the lookup list appears.

**Comment by levalencia** [comment]
have you tested this on subsites?

**Comment by patrick-rodgers** [comment]
Closing this issue due to age. If you believe this is still an issue in the latest release please re-open and let us know.

**Comment by KEMiCZA** [comment]
@patrick-rodgers  It seems that this issue is still not fixed.
![capture](https://cloud.githubusercontent.com/assets/3862716/12331670/d082c5a0-baea-11e5-9549-2d2238da2a9c.PNG)
![xml](https://cloud.githubusercontent.com/assets/3862716/12331672/d22ff166-baea-11e5-8147-1d6d839439cf.PNG)

**Comment by jansenbe** [comment]
I've been testing this and I'm able to provision a lookup field, including multiple columns as shown in below screenshot:
![image](https://cloud.githubusercontent.com/assets/7451219/22380222/ea452418-e4bc-11e6-8c23-bd53d14e521c.png)

This has been realized using the fields defined in this template: https://github.com/SharePoint/PnP-Sites-Core/blob/master/Core/OfficeDevPnP.Core.Tests/Framework/Functional/Templates/field_add.xml

I'm closing this issue for now, if it's still not possible to correctly provision lookup fields using the approach mentioned above I would suggest to open a new issue, provide a clean repro and explain the used test environment.

**Comment by NPrice99** [comment]
Hi Bert

On my SharePoint Tenancy I got this error :-

Apply-SPOProvisioningTemplate : Invalid field name. {3a18fcce-7eab-45d7-b6b6-c61cec868057}
https://njpenterprises.sharepoint.com/sites/dev
At line:1 char:1
+ Apply-SPOProvisioningTemplate -Path "c:\temp\lookup-template.xml"
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Apply-PnPProvisioningTemplate], ServerException
    + FullyQualifiedErrorId : EXCEPTION,SharePointPnP.PowerShell.Commands.Provisioning.ApplyProvisioningTemplate



I enclose the debug output.

Regards

Nigel

**Comment by jansenbe** [comment]
Would recommend testing with dev build (or with Feb release in 2 weeks), but on the other hand this xml we use for testing has been working for months. You need to specify both termgroup and field handlers as shown in the test code: https://github.com/SharePoint/PnP-Sites-Core/blob/dev/Core/OfficeDevPnP.Core.Tests/Framework/Functional/Implementation/FieldImplementation.cs#L19. Here you can see the last test run for SP2013 for this test: https://testautomation.sharepointpnp.com/Testcase/Details/1359674?branch=dev

Important note: this test assumes there's a library named documents in the site collection

**Comment by NPrice99** [comment]
Hi Bert

I was using Erwin’s PnP PowerShell commands to test this, not C#.

I know it should not matter.

The error message was that the template had an invalid “Field Name”.  I don’t think it had got as far as the field handler.

I have added the handlers parameters to the line of PnP PowerShell and still got this :-

PS C:\Windows\system32> Apply-SPOProvisioningTemplate -Path "c:\temp\lookup-template.xml" -Handlers Fields,TermGroups
Apply-SPOProvisioningTemplate : Invalid field name. {3a18fcce-7eab-45d7-b6b6-c61cec868057}
https://njpenterprises.sharepoint.com/sites/dev
At line:1 char:1
+ Apply-SPOProvisioningTemplate -Path "c:\temp\lookup-template.xml" -Ha ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Apply-PnPProvisioningTemplate], ServerException
    + FullyQualifiedErrorId : EXCEPTION,SharePointPnP.PowerShell.Commands.Provisioning.ApplyProvisioningTemplate

Regards

Nigel
